Author: Daniel F. Dickinson <cshored@thecshore.com>
Date:   Wed Jan 17 03:09:48 2018 -0500

    mountd: Add patch to prevent endian.h from throwing warnings with musl

    Due to -Werror mountd experiences build failures due to warnings
    about needing parentheses in endian.h imported from sys/types.h

    In the files where sys/types.h is included removed the explicit
    include solves the isues as it is better handled as an internal
    include for auto_fs4.h.

Signed-of-by: Daniel F. Dickinson <cshored@thecshore.om>
Index: git/autofs.c
===================================================================
--- git.orig/autofs.c
+++ git/autofs.c
@@ -1,4 +1,3 @@
-#include <sys/types.h>
 #include <linux/types.h>
 #include <paths.h>
 #include <time.h>
@@ -14,7 +13,6 @@
 #include <syslog.h>
 #include <unistd.h>
 #include <sys/ioctl.h>
-#include <sys/types.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <sys/time.h>
Index: git/fs.c
===================================================================
--- git.orig/fs.c
+++ git/fs.c
@@ -2,7 +2,6 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
 
Index: git/main.c
===================================================================
--- git.orig/main.c
+++ git/main.c
@@ -2,7 +2,6 @@
 #include <string.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include <sys/types.h>
 
 #include "include/log.h"
 #include "include/sys.h"
Index: git/mount.c
===================================================================
--- git.orig/mount.c
+++ git/mount.c
@@ -3,7 +3,6 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <sys/stat.h>
-#include <sys/types.h>
 #include <fcntl.h>
 #include <sys/ioctl.h>
 #include <linux/hdreg.h>
@@ -12,7 +11,6 @@
 #include <sys/wait.h>
 #include <sys/inotify.h>
 #include <sys/stat.h>
-#include <sys/types.h>
 #include <glob.h>
 #include <libgen.h>
 #include <poll.h>
